lib/datagrid/core.rb

```diff
@@ -266,7 +270,13 @@ def reset
     protected
 
     def sanitize_for_mass_assignment(attributes)
-      forbidden_attributes_protection ? super : attributes
+      if forbidden_attributes_protection
+        super
+      elsif defined?(ActionController::Parameters) && attributes.is_a?(ActionController::Parameters)
+        attributes.permit!.to_h
```

Comment:
There’s a dedicated method for this: https://apidock.com/rails/ActionController/Parameters/to_unsafe_h
```suggestion
        attributes.to_unsafe_h
```